lib/web_ui/lib/src/engine/skwasm/skwasm_impl/shaders.dart

```diff
@@ -155,8 +155,43 @@ class SkwasmGradient extends SkwasmShader implements ui.Gradient {
   }
 }
 
+class SkwasmImageShader extends SkwasmShader implements ui.ImageShader {
+  SkwasmImageShader._(this.handle);
+
+  factory SkwasmImageShader.imageShader(
+    SkwasmImage image,
+    ui.TileMode tmx,
+    ui.TileMode tmy,
+    Float64List matrix4,
+    ui.FilterQuality? filterQuality,
+  ) => withStackScope((StackScope scope) {
+    final RawMatrix33 localMatrix = scope.convertMatrix4toSkMatrix(matrix4);
```

Comment:
Since the matrix is never null, can we allocate on the heap to avoid copying the matrix twice, once here and a second time in `shader_createFromImage`?

Reply:
Two stack allocations is likely cheaper than one heap allocation. Also, I'm not entirely sure it really wins anything to do things on the heap. `SkMatrix` is actually not a POD, so you have to actually call the constructor on it (we can't just try to fake construction of it through memory accesses). I wouldn't worry about it much, this is likely to be very fast.

Reply:
Gotcha. I thought the second one was on the heap.